inc/cropper/namespace.php

```diff
@@ -488,3 +507,370 @@ function prevent_thumbnail_generation( $sizes ) {
+		return $data;
+	}
+
+	$data = massage_meta_data_for_orientation( $data );
+
+	// Full size image info.
+	$image_sizes = get_image_sizes();
+	$mime_type = get_post_mime_type( $attachment_id );
+	$filename = pathinfo( $data['file'], PATHINFO_FILENAME );
+	$ext = pathinfo( $data['file'], PATHINFO_EXTENSION );
+	$orig_w = $data['width'];
+	$orig_h = $data['height'];
+
+	foreach ( $image_sizes as $size => $crop ) {
+		if ( isset( $data['sizes'][ $size ] ) ) {
+			// Meta data is set.
+			continue;
+		}
+
+		if ( 'full' === $size ) {
+			// Full is a special case.
+			continue;
+		}
+
+		/*
+		 * $new_dims = [
+		 *    0 => 0
+		 *    1 => 0
+		 *    2 => // Crop start X axis
+		 *    3 => // Crop start Y axis
+		 *    4 => // New width
+		 *    5 => // New height
+		 *    6 => // Crop width on source image
+		 *    7 => // Crop height on source image
+		 * ];
+		*/
+		$new_dims = image_resize_dimensions( $orig_w, $orig_h, $crop['width'], $crop['height'], $crop['crop'] );
+
+		if ( ! $new_dims ) {
+			continue;
+		}
+
+		$w = (int) $new_dims[4];
+		$h = (int) $new_dims[5];
+
+		// Set crop hash if source crop isn't 0,0,orig_width,orig_height
+		$crop_details = "{$orig_w},{$orig_h},{$new_dims[2]},{$new_dims[3]},{$new_dims[6]},{$new_dims[7]}";
+		$crop_hash = '';
+
+		if ( $crop_details !== "{$orig_w},{$orig_h},0,0,{$orig_w},{$orig_h}" ) {
+			/*
+			 * NOTE: Custom file name data.
+			 *
+			 * The crop hash is used to help determine the correct crop to use for identically
+			 * sized images.
+			 */
+			$crop_hash = '-c' . substr( strtolower( sha1( $crop_details ) ), 0, 8 );
+		}
+
+		// Add meta data with fake WP style file name.
+		$data['sizes'][ $size ] = [
+			'width' => $w,
+			'height' => $h,
+			'file' => "{$filename}{$crop_hash}-{$w}x{$h}.{$ext}",
+			'mime-type' => $mime_type,
+		];
+	}
+
+	$cache[ $attachment_id ] = $data;
+	return $data;
+}
+
+/**
+ * Swap width and height if required.
+ *
+ * The Tachyon service/sharp library will automatically fix
+ * the orientation but as a result, the width and height will
+ * be the reverse of that calculated on upload.
+ *
+ * This swaps the width and height if needed but it does not
+ * fix the image on upload as we have Tachy for that.
+ *
+ * @param array $meta_data Meta data stored in the database.
+ *
+ * @return array Meta data with correct width and height.
+ */
+function massage_meta_data_for_orientation( array $meta_data ) {
+	if ( empty( $meta_data['image_meta']['orientation'] ) ) {
+		// No orientation data to fix.
+		return $meta_data;
+	}
+
+	$fix_width_height = false;
+
+	switch ( $meta_data['image_meta']['orientation'] ) {
+		case 5:
+		case 6:
+		case 7:
+		case 8:
+		case 9:
+			$fix_width_height = true;
+			break;
+	}
+
+	if ( ! $fix_width_height ) {
+		return $meta_data;
+	}
+
+	$width = $meta_data['height'];
+	$meta_data['height'] = $meta_data['width'];
+	$meta_data['width'] = $width;
+	return $meta_data;
```

Comment:
```suggestion
	unset( $meta_data['image_meta']['orientation'] );
	return $meta_data;
```

This will prevent the problem returning if multiple filters like this run.